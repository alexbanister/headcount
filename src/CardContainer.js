import React from 'react';
import Card from './Card';
import PropTypes from 'prop-types';

const CardContainer = (
  {districtData, districtList,
    setComparePosition, firstDistrict, secondDistrict} ) => {
  const allCards = districtList.map( (place) => {
    let selectedClass = '';
    if (place === firstDistrict || place === secondDistrict) {
      selectedClass = 'selected';
    }
    return <Card
      selected={selectedClass}
      setComparePosition={setComparePosition}
      district={districtData.findByName(place)}
      key={place}/>;
  });

  return (
    <div className="card-container">
      {allCards}
    </div>
  );
};

CardContainer.propTypes = {
  districtData: PropTypes.objectOf(PropTypes.object),
  districtList: PropTypes.arrayOf(PropTypes.string),
  setComparePosition: PropTypes.func,
  firstDistrict: PropTypes.string,
  secondDistrict: PropTypes.string
};


export default CardContainer;
