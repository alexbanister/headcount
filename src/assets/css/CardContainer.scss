@import "variables.scss";

.card-container {
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
}
