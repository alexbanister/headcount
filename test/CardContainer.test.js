import React from 'react';
import { shallow } from 'enzyme';
import CardContainer from '../src/CardContainer';
import DistrictRepository from '../src/helper.js';
import kinderData from '../data/kindergartners_in_full_day_program.js';

describe('CardContainer', () => {
  const district = new DistrictRepository(kinderData);

  it('should create an instance of a card container', () => {
    const mockFunc = jest.fn();
    const wrapper = shallow(<CardContainer
      firstDistrict={'COLORADO'}
      secondDistrict={'ACADEMY 20'}
      setComparePosition={mockFunc}
      districtData={district.data}
      districtList={[]}
    />);

    expect(wrapper).toMatchSnapshot();
  });
});
