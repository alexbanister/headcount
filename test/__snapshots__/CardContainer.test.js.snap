exports[`CardContainer should create an instance of a card container 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardContainer
    districtData={undefined}
    districtList={Array []}
    firstDistrict="COLORADO"
    secondDistrict="ACADEMY 20"
    setComparePosition={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
      "className": "card-container",
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
        "className": "card-container",
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
